docs/Architecture Decision Record/013-download-strategy.md

```diff
@@ -0,0 +1,38 @@
+# 13. Download Strategy
+
+Date: 2021-04-06 (yyyy-mm-dd)
+
+## Status
+
+Approved
+
+## Context
+
+In [this issue](https://github.com/raft-tech/TANF-app/issues/771) we are investigating the use of pre-signed URLs to determine whether there are security issues with the approach.
+
+We had originally implemented pre-signed URLs for downloading files because the system would need to download the files from S3 and then download them again to the client, using resources on the backend for every download. This would not cause a problem in this stage of development, but when the backend will be charged with parsing data from potentially large files, those system resources would become more precious. Using pre-signed URLs takes the added pressure off of the backend entirely.
+
+## Decision
+
+We believe the use of time/IP address limited signed URLs is a reasonably secure approach to downloading files from S3. However, we also believe that it may cause issues with our ATO approval as the data is highly sensitive. Furthermore, 18F published a recommendation today, [recommending to not use pre-signed URLs](https://engineering.18f.gov/security/cloud-services/) for FISMA High projects.
+
+In our investigation we discovered a way that we can securely download the files from the backend while streaming the files directly from S3 to the client, taking any pressure off of resources needed for parsing files on the backend. 
```

Comment:
@carltonsmith sounds promising! can we include reference links here to what was found?

Reply:
The feature isn't documented in the docs, but can be found in the codebase, which is where we found it. I can include a link but it takes a bit of analysis to understand what is happening.

Reply:
@carltonsmith, yes it'd be great to have the link but also, i think it'd be really helpful to have a walk-through of part of the codebase where this is happening. let us know when you think it might be best to do this (if not today); or happy to have your suggestions on other ways we can go about learning about this alternative as part of qasp review 😃

Reply:
Updated this PR with technical documentation to explain how the downloads will work under this strategy.

Pasting the associated flowchart here for reference:
![tdp-data-file-download-api](https://user-images.githubusercontent.com/22626085/115057902-9b315180-9eb2-11eb-9da0-109d34ca8493.png)